app/lib/actions/katello/event_queue/monitor.rb

```diff
@@ -0,0 +1,60 @@
+module Actions
+  module Katello
+    module EventQueue
+      class Monitor < Actions::EntryAction
+        include ::Dynflow::Action::Singleton
+
+        Handle = Algebrick.type do
+          fields! event_type: String, object_id: Integer
+        end
+
+        Poll = Algebrick.atom
+
+        def self.launch(world)
+          unless self.singleton_locked?(world)
+            ForemanTasks.trigger(self)
+          end
+        rescue StandardError
+          # TODO: what to do here?
+          Rails.logger.info("Already started!")
+        end
+
+        def plan
+          plan_self
+        end
+
+        def run(event = nil)
```

Comment:
@adamruzicka can you take a look at this class to see if it's a proper implementation of an event-based singleton action? Anything that I've missed?